python/cudf/cudf/core/buffer.py

```diff
@@ -84,6 +102,15 @@ def _init_from_array_like(self, data):
                 f"Cannot construct Buffer from {data.__class__.__name__}"
             )
 
+    def serialize(self):
+        header = self.__cuda_array_interface__.copy()
```

Comment:
Looks like this header isn't used in deserialization, is it used by distributed or ucx-py?

Reply:
Yeah now that I think about it. We can probably drop the `serialize`/`deserialize` methods from `Buffer` and drop the addition of their headers.

Thought it might be useful in building `ColumnBase` or other things to have this information available for verification potentially. Like making sure we got the size expected. However this already is done in Distributed and UCX-Py. So we don't strictly need to do that here too.

Reply:
FWIW I pushed a commit with validation just so we could see what it looked like. Happy to leave it or drop it and the headers.

Reply:
We could consider other checks here, but maybe they make more sense as checks on `Buffer` objects generally. ( https://github.com/rapidsai/cudf/issues/4104 )

Reply:
Added a little bit of logic in the header to aid serializing `Buffer` subclasses like cuML's `Array`. ( https://github.com/rapidsai/cuml/pull/1635 )

cc @dantegd